plugin/evm/vm_test.go

```diff
@@ -3926,3 +3934,169 @@ func TestCreateHandlers(t *testing.T) {
 		require.ErrorIs(t, elem.Error, rpc.ErrMissingBatchResponse)
 	}
 }
+
+// TestBlockGasValidation tests the two validation checks:
+// 1. invalid gas used relative to capacity
+// 2. total intrinsic gas cost is greater than claimed gas used
+func TestBlockGasValidation(t *testing.T) {
+	newBlock := func(
+		t *testing.T,
+		vm *VM,
+		claimedGasUsed uint64,
+	) *types.Block {
+		require := require.New(t)
+
+		chainExtra := params.GetExtra(vm.chainConfig)
+		parent := vm.eth.APIBackend.CurrentBlock()
+		const timeDelta = 5 // seconds to fill gas capacity
```

Comment:
Nit: Subnet-EVM doesn't have ACP-176 yet, so there is no notion of "gas capacity". I think this is really just a mocked block delay in this case